src/java/za/co/mip/ablduck/utilities/HTMLGenerator.java

```diff
@@ -0,0 +1,311 @@
+
+import za.co.mip.ablduck.models.SourceJSObject;
+import za.co.mip.ablduck.models.generic.DeprecatedObject;
+import za.co.mip.ablduck.models.generic.MetaObject;
+import za.co.mip.ablduck.models.source.MemberObject;
+import za.co.mip.ablduck.models.source.ParameterObject;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.text.MessageFormat;
+
+public class HTMLGenerator {
+    private static final ResourceBundle TEMPLATES = ResourceBundle.getBundle("za.co.mip.ablduck.utilities.templates");
+    
+    private static Pattern returnType = Pattern.compile("\\):(.*)");
+    private Map<String, SourceJSObject> classes;
+    
+    public String getTemplate(String key) {
+        return TEMPLATES.getString(key);
+    }
+            
+    public String getClassHtml (Map<String, SourceJSObject> allclasses, SourceJSObject cls) {
+        this.classes = allclasses;
+        
+        return MessageFormat.format(getTemplate("CLASSBODY"), 
+                                            renderSidebar(cls),
+                                            renderClassComment(cls),
+                                            renderMemberDetails(cls, "property", "Properties"),
+                                            renderMemberDetails(cls, "method", "Methods"));
+    }
+    
+    private String renderSidebar (SourceJSObject cls) {
+        if(!"".equals(cls.ext) || !cls.subclasses.isEmpty())
+            return MessageFormat.format(getTemplate("SIDEBAR"), 
+                                            renderClassTree(cls),
+                                            renderSubclasses(cls),
+                                            renderAuthor(cls));
+        
+        return "";
+    }
+    
+    private String renderAuthor (SourceJSObject cls) {
+        if (!"".equals(cls.author))
+            return MessageFormat.format(getTemplate("AUTHOR"), cls.author);
+                                                
+        return "";
+    }
+    
+    private String renderClassTree (SourceJSObject cls) {
+        if(!cls.superclasses.isEmpty()) {
+            //Add this class to the tree for display purposes
+            cls.superclasses.add(cls.name);
+        
+            String classTree = renderSuperTree(cls.superclasses, 0);
+            
+            return MessageFormat.format(getTemplate("HIERARCHY"), classTree);
+        }
+        
+        return "";
+    }
+    
+    private String renderSuperTree (List<String> superclasses, Integer i) {
+        //If we are finished rendering the links leave
+        if (i == superclasses.size())
+            return "";
+        
+        //Add the first child class if we are the first class
+        String cssClass = "";
+        if(i==0)
+            cssClass = "first-child";
+        
+        //Render superclasses as a hyperlink unless we are the first class, ourself
+        String classLink = renderLink(superclasses.get(i).toString());

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') "get" returns a string, there's no need to call "toString()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1858)